spinalcordtoolbox/math.py

```diff
@@ -113,14 +113,56 @@ def dilate(data, size, shape, dim=None):
     """
     if isinstance(data, Image):
         im_out = data.copy()
-        im_out.data = dilate(data.data, size, shape, dim)
+        im_out.data = dilate(data.data, size, shape, dim, islabel)
         return im_out
     else:
         footprint = _get_footprint(shape, size, dim)
-        if data.dtype in ['uint8', 'uint16']:
-            return rank.maximum(data, footprint=footprint)
+        if islabel:
+            return _dilate_point_labels(data, shape=shape, size=size, footprint=footprint)
         else:
-            return dilation(data, footprint=footprint, out=None)
+            if data.dtype in ['uint8', 'uint16']:
+                return rank.maximum(data, footprint=footprint)
+            else:
+                return dilation(data, footprint=footprint, out=None)
+
+
+def _dilate_point_labels(data, shape, size, footprint):
+    """
+    A more efficient dilation algorithm when we know the image is mostly nonzero (i.e. point label image)
+    """
```

Comment:
I might have done a lot of work for nothing? After some digging, it seems like scipy's `grey_dilation` may be significantly faster than skimage's dilation: https://github.com/scikit-image/scikit-image/issues/1190#issuecomment-572384985

Meaning, we don't have to reinvent the wheel here...

Reply:
EDIT: Nope, `grey_dilation` is still as slow as numpy's `dilation` in my tests. I wonder why?